pkg/cache/cache.go

```diff
@@ -148,6 +178,65 @@ func (c *Cache) newClusterQueue(cq *kueue.ClusterQueue) (*ClusterQueue, error) {
 	return cqImpl, nil
 }
 
+func (c *Cache) WaitForPodsReady(ctx context.Context) {
+	c.Lock()
+	defer c.Unlock()
+
+	log := ctrl.LoggerFrom(ctx)
+	for {
+		if c.podsReadyForAllAdmittedWorkloads(ctx) {
+			return
+		}
+		log.V(3).Info("Blocking admission as not all workloads are in the PodsReady condition")
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			c.podsReadyCond.Wait()
+		}
+	}
+}
+
+func (c *Cache) podsReadyForAllAdmittedWorkloads(ctx context.Context) bool {
+	log := ctrl.LoggerFrom(ctx)
+	for _, cq := range c.clusterQueues {
+		for _, wl := range cq.Workloads {
```

Comment:
Something to think about: what if we track "not ready" workloads in each ClusterQueue, and so instead of iterating over all workloads every time, we just check if that list is empty? This will also make it not expensive to call `podsReadyForAllAdmittedWorkloads`, and so we can remove the checks that we have around broadcast.

Reply:
+1, although I'm fine leaving this as a follow up, to focus on the semantics first.

Reply:
Done, I think it also simplifies the code.